httpcache_test.go

```diff
@@ -18,6 +18,7 @@ var s struct {
 	server    *httptest.Server
 	client    http.Client
 	transport *Transport
+	infChan   chan struct{}
```

Comment:
It'd help to leave an inline comment explaining `infChan`, what it's for and what it's doing.

Also, from the way it's being used, can I suggest a better, more general name `done`?

```Go
done chan struct{} // Closed when test is done.
```

That'd be more clear because it's analogous with `context` package's `Done` channel.